(DO NOT MERGE) Update machine auth docs after new additions (#2560)
clerkinc/clerk-docs: docs/references/nuxt/overview.mdx

The `Auth` object is available at `event.context.auth()` in your [event handlers](https://h3.unjs.io/guide/event-handler). This JavaScript object contains important information like session data, your user's ID, as well as their organization ID. [Learn more](/docs/references/backend/types/auth-object).

### `event.context.auth()` options

<Properties>
  - `opts?`
  - `{acceptsToken: TokenType, treatPendingAsSignedOut: boolean }`

  An optional object that can be used to configure the behavior of the `event.context.auth()` function. It accepts the following properties:

  <Include src="_partials/machine-auth/accepts-token-explanation" />

  - `treatPendingAsSignedOut?`: A boolean that indicates whether to treat [`pending` session status](/docs/references/javascript/types/session-status#properties) as signed out. Defaults to `true`.
</Properties>

> ### Example: Protecting a route by accepting and verifying multiple token types using `event.context.auth()`

Reviewer: ```suggestion
### Example: Protect a route based on token type
```

The following example uses `event.context.auth()` to protect the route based on token type:

- It accepts any token type `(acceptsToken: 'any')` from the request.
- If the token is a `session_token`, it logs that the request is from a user session.
- Otherwise, it logs that the request uses a machine token and specifies its type.

```ts
export default eventHandler((event) => {
  const auth = event.context.auth({ acceptsToken: 'any' })

  if (authObject.tokenType === 'session_token') {
    console.log('This is a session token from a user')
  } else {

>     console.log('This is some other type of machine token (api_key | oauth_token | m2m_token)')

Reviewer: same here
